### ISSUE TITLE
[v3] Implement new Concept Exercise: options

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#860]
This issue describes how to implement the `options` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Options in [F#][options].

## Learning objectives

- Know of the existing of the `Option<T>` type and when to use it.
- Know how to define an option value.
- Know how to use pattern matching with options.
- Know of some basic functions in the `Option` module (like `Option.map`) to modify options.
- Know how options differ from `null`.
- Know what generic types are.

## Out of scope

- Custom computation expression to deal with options.
- Custom operators to deal with options applicatively (like `<*>`).
- Converting to and from other types (like arrays of lists).

## Concepts

The Concepts this exercise unlocks are:

- `options`: know of the existing of the `Option<T>` type and when to use it; know how to define an optional value; know of some basic functions in the `Option` module (like `Option.map`) to modify options; know how options differ from `null`; know how to use pattern matching with options.
- `generic-types`: know what generic types are.

## Prerequisites

This exercise's prerequisites Concepts are:

- `discriminated-unions`: know that an option is implemented as a DU and know how to do pattern matching on a DU.
- `higher-order-functions`: how to use higher-order functions.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Options][options]: shows how options in F# work and how to use them.
- [Options introduction][options-introduction]: nice introduction to using options.

### After

- [Options][options]: shows how options in F# work and how to use them.
- [Options introduction][options-introduction]: nice introduction to using options.
- [C# null vs F# Option][options-versus-null]: introdution to options in F#, but from the viewpoint of a C# user.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/fsharp/reference/README.md#reference-docs
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[options]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/options
[options-introduction]: https://fsharpforfunandprofit.com/posts/the-option-type/
[options-versus-null]: http://www.garethrepton.com/Fsharp-handling-the-option-type/

**Comment by Grenkin1988** [comment]
@ErikSchierboom Hello, as I understand this is this open for implementation? Can I take it? Does any post on forum required?
